cmd/soroban-cli/src/commands/contract/invoke.rs

```diff
@@ -189,7 +189,17 @@ impl Cmd {
             .map(|i| {
                 let name = i.name.to_string().unwrap();
                 if let Some(mut raw_val) = matches_.get_raw(&name) {
-                    let s = raw_val.next().unwrap().to_string_lossy().to_string();
+                    let mut s = raw_val.next().unwrap().to_string_lossy().to_string();
+                    if matches!(i.type_, ScSpecTypeDef::Address) {
+                        let cmd = crate::commands::config::identity::address::Cmd {
+                            name: Some(s.clone()),
+                            hd_path: Some(0),
```

Comment:
Unrelated side note: Been thinking about this hd-path thing more, and it feels like a sort of leaky abstraction in identities. What if it was rolled into a named identity thing? So, instead of passing `--hd-path` on the CLI you'd first have to do:
```
# Explicitly generate a derived subpath key for usage, something like
$ soroban config identity generate --parent existing_identity --hd-path 1 child_name

# and then you'd use it like a normal identity
$ soroban contract invoke --source child_name ...
```
Thoughts?

Reply:
I was thinking that we could also do something like `existing_id:<hd_path>` then that id file can just reference the other id.